### PR TITLE
change default dart2js args to the new -O2

### PR DESCRIPTION
**Pull request by jakemac53**
This arg is intended to contain all "safe" optimizations, which today includes `--minify` and `--lax-runtime-type-to-string`.

### CODE REVIEW

**Comment by jakemac53** [comment]
Making it a special option would require checking for a conflicting arg in dart2js_args though... but it would make it harder to accidentally override it.

**Comment by natebosch** [comment]
Yeah, we might even be able to get away with not checking for a conflict.

I just think this one is special enough that's it's worth supporting directly. With config set alias it'll also make it possible to do with a command line flag to the build system

File: build_web_compilers/pubspec.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: build_web_compilers
-version: 0.4.3
+version: 0.4.4-dev
```
Comment (kevmoo) [inline build_web_compilers/pubspec.yaml]: Don't we need to update the minimum SDK constraint to require `2.1.0-dev..??` for when this was added? It's not in dart2 stable

Comment (jakemac53) [reply build_web_compilers/pubspec.yaml]: Good catch!

**Comment by jakemac53** [comment]
Going to just close this for now, and wait on a 2.1.0 release as I would prefer to not have the minimum bound be a dev release.

**Comment by kevmoo** [comment]
@jakemac53 cleanup this branch?

**Pull request closed without merging**